text/0006-equivocation-catastrophes.md

```diff
@@ -0,0 +1,114 @@
+
+The first essential step will be determining the starting point for the network after the restart. The suggested approach here would be to agree upon the new initial global state, new set of validators and restart the network as if it was starting from scratch.
+
+A desirable goal for the initial global state would be to revert as few transactions from all the forks as possible, in order to limit the damage. Every transaction that was included in a block which appeared to be finalized for a time could have caused real world consequences (like, for example, a merchant sending ordered goods). Reverting any such transaction might potentially cause a loss to a user.
+
+In order to minimize losses, the following approach is proposed: first, choose the last safe block as the starting point. A safe block would be one that still appears finalized with some small FTT, for example 1%, after a full synchronization of the protocol state.
+
+After the block is chosen, take its global state and all transactions from the blocks that were considered finalized at some point, but are no longer a part of the main branch of the chain, and attempt to replay them on top of that global state. Since the order could matter, the community would have to agree on one (the default one could be based on deploy timestamps, but the community could decide on some other ordering if deemed necessary). The deploys' TTL could also be expanded to account for the time spent reaching social consensus and allow them to still be valid afterwards. Call global state resulting from replaying these deploys the "pre-initial state".
+
+As the last step, agree on the initial set of validators for the restarted network. The equivocators should be slashed, but there may be reasons to also exclude other validators, or even to start with a set of validators that has nothing in common with the validators from the stopped network. Once agreement has been reached, apply necessary changes to the pre-initial state: modify the validator stakes and change the entries in the auction contract. After this is applied, the resulting state will be the state used for the restart.
+
+After the users collectively decide upon the new initial state and the new initial set of validators, the network can be restarted.
+
+Note that tools would have to be provided that would facilitate the manual construction of a specific state. This could be realized in the form of node software constructing the state based on the provided hash of a block, list of block or deploy hashes defining deploys to be executed on top of that block, and possibly some additional upgrade code.
+
+### Restarting the network
```

Comment:
This section should also include, mentioned earlier, set of deploys for replay/reinclusion.

Reply:
The deploy hashes are mentioned - or do you mean that we should explicitly state that the deploy _bodies_ are needed as well?

Reply:
I meant in the list below – we somehow have to get that information into the node.